discover.py

```diff
@@ -0,0 +1,234 @@
+import logging
+import time
+import shutil
+import subprocess
+from threading import Thread
+from datetime import datetime
+
+from zeroconf import ServiceBrowser, Zeroconf, ServiceInfo
+import netifaces
+from flask import Flask
+from flask_jsontools import JsonSerializableBase, jsonapi
+from waitress import serve
+
+import supervisor_client
+
+log = logging.getLogger('discovery')
+
+SERVICE_TYPE = "_liquid._tcp.local."
+HOST_ONLY_NETWORK_MASK = '255.255.255.255'
+DNSMASQ_PROCESS_NAME = 'dnsmasq-dns'
+
+nodes = {}
+
+def reload_dnsmasq():
+    log.debug("Reloading dnsmasq-dns")
+    supervisor = supervisor_client.connect_unix_socket()
+    is_running = supervisor.getProcessInfo(DNSMASQ_PROCESS_NAME)['statename'] == 'RUNNING'
+    if is_running:
+        supervisor.stopProcess(DNSMASQ_PROCESS_NAME)
+    supervisor.startProcess(DNSMASQ_PROCESS_NAME)
+
+def stop_dnsmasq():
+    log.debug("Stopping dnsmasq-dns")
+    supervisor = supervisor_client.connect_unix_socket()
+    supervisor.stopProcess(DNSMSAQ_PROCESS_NAME)
+
+def rewrite_dnsmasq_conf(path, conf_string):
+    temp = path + ".tmp"
+    with open(temp, 'w') as f:
+        f.write(conf_string)
+    shutil.move(temp, path)
+
+class DnsmasqRestarter(Thread):
```

Comment:
Are we still subclassing `Thread`? I think it's cleaner to have a function which [gets called by threading](https://docs.python.org/3/library/threading.html#threading.Thread):

```python
from threading import Thread
def update_loop():
    while True:
        do_stuff()
Thread(target=update_loop).start()
```